Utils/trigger_private_build.py

```diff
@@ -0,0 +1,162 @@
+
+
+def get_modified_files(branch_name: str = None) -> List[str]:
+    """ Gets modified files between master branch and the input branch_name, If the branch_name is empty the method
+        compare master branch with the commit sha1 from the environment variable CIRCLE_SHA1.
+
+    Args:
+        branch_name: The branch name to compare with master.
+
+    Returns: A list of modified files.
+
+    """
+    if not branch_name:
+        branch_name = os.environ.get('CIRCLE_SHA1')
+
+    files = []
+    files_string = tools.run_command(f'git diff --name-only origin/master...{branch_name}')
+    for line in files_string.split("\n"):
+        if line:
+            files.append(line)
+    return files
+
+
+def branch_has_private_build_infra_change(branch_name: str = None) -> bool:
+    """ Checks whether the modified files in the branch are private build infrastructure files.
+
+    Args:
+        branch_name: The branch name to compare with master.
+
+    Returns: True if private build infrastructure files modified, False otherwise.
+
+    """
+    modified_files = get_modified_files(branch_name)
+    for infra_file in modified_files:
+        if infra_file in PRIVATE_BUILD_INFRA_SCRIPTS:
+            return True
+
+        path = os.path.dirname(infra_file)
+        for infra_code_dir_path in PRIVATE_BUILD_INFRA_FOLDERS:
+            if path.startswith(infra_code_dir_path):
+                return True
+    return False
+
+
+def get_dispatch_workflows_ids(github_token: str, branch: str) -> List[int]:
+    """ Gets private repo dispatch workflows on the given branch.
+
+    Args:
+        github_token: Github bearer token.
+        branch: The branch to get the workflows from.
+
+    Returns: A list of workflows ids.
+
+    """
+    res = requests.get(GET_DISPATCH_WORKFLOWS_URL,
+                       headers={'Authorization': f'Bearer {github_token}'},
+                       params={'branch': branch, 'event': 'repository_dispatch'},
+                       verify=False)
+    if res.status_code != 200:
+        logging.error(f'Failed to get private repo workflows, request to '
+                      f'{GET_DISPATCH_WORKFLOWS_URL} failed with error: {str(res.content)}')
+        sys.exit(1)
+
+    try:
+        workflows = json.loads(res.content)
+    except ValueError:
+        logging.error('Enable to parse private repo workflows response')
+        sys.exit(1)
+
+    workflows = workflows.get('workflow_runs', [])
+    return [workflow.get('id') for workflow in workflows]
+
+
+def main():
+    install_logging("TriggerPrivateBuild.log")
+    # get github token and commit sha1 parameters
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument('--github-token', help='Github token')
+    arg_parser.add_argument('--commit-sha1', help='commit sha1 for creating the private repo')
+    args = arg_parser.parse_args()
+
+    github_token = args.github_token
+    branch_name = os.environ.get('CIRCLE_BRANCH')
```

Comment:
What about he CIRCLE_SHA1?

Reply:
@hod-alpert I updated the code to gets the the `commit-sha1` as argument to `trigger_private_build.py` script, the `branch_name` used in the event name only:
<img src="https://user-images.githubusercontent.com/46249224/109960094-00f5bf80-7cf1-11eb-8aec-f6db783cfdcb.png" width="400" />